src/Symfony/Bundle/FrameworkBundle/Command/EventDispatcherDebugCommand.php

```diff
@@ -59,12 +90,24 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        // BC to be removed in 4.0
+        if (__CLASS__ !== get_class($this)) {
+            $r = new \ReflectionMethod($this, 'getEventDispatcher');
+            if (__CLASS__ !== $r->getDeclaringClass()->getName()) {
+                @trigger_error(sprintf('Usage of method "%s" is deprecated since version 3.4 and will no longer be supported in 4.0.', get_class($this).'::getEventDispatcher'), E_USER_DEPRECATED);
```

Comment:
```diff
- 'Usage of method "%s" is deprecated since version 3.4 and will no longer be supported in 4.0.'
+ 'Usage of method "%s" is deprecated since version 3.4 and will no longer be supported in 4.0. Inject an event dispatcher instance using the constructor instead.'
```

?